### DISCUSSION TITLE
How to make MediaPipeFaceLink "see" NvAR dll files?

### DISCUSSION THREAD

**Discussion by GreebleGraphics** [endink/Mediapipe4u-plugin#97] [Q&A] [answered]
Greetings,

I have the latest version of the `MediaPipe4U-Demo`, and the older `M4U_UE5_1_20230718.7z` version of the MediaPipe4U plugin, which works with the demo project. I am trying to figure out how to get the demo project to "see" the NvAR dll files.

I have the `NVIDIA+AR+SDK_0.8.2.0` version of the NVIDIA Maxine AR SDK. At first, the `MediaPipeFaceLink` level could not find the "NVCVImage.dll" file. I added the folder path of the "bin" folder that contains "NVCVImage.dll" to my PATH environment variable. I am now getting this error:

`Error        LogMediaPipeNvAR          [NvAR API] The file could not be found ( code: NVCV_ERR_FILE, op: NvAR_Load)`

What is the proper way to set up NvAR for MediaPipe4U?

I appreciate any advice. I will continue looking in the configs, the code, and the documentation.

Thanks,

Jen

**Comment by endink** [comment]
Have you seen this?

https://opensource-labijie-com.translate.goog/Mediapipe4u-plugin/mocap/experiment/nvar.html?_x_tr_sl=zh-CN&_x_tr_tl=en&_x_tr_hl=zh-CN&_x_tr_pto=wapp

**Comment by endink** [comment]
You can't use nvidia demo on github, you have to install NVIDIA AR with an installer, in my experience, the SDK files for different GPU may be different. 

For example, the SDKs for 2060 and 3060 are different

**Comment by GreebleGraphics** [comment, marked as answer]
Thank you so much for the swift replies and clear instructions. I took these steps based on your information, and I got it working:

1. I went to the [NVIDIA Broadcast Download Center](https://www.nvidia.com/en-us/geforce/broadcasting/broadcast-sdk/resources/) and downloaded the 1GB AR installer for my specific RTX GPU
2. I ran the installer
3. I launched the MediaPipe4U-Demo project in the Unreal Editor
4. I loaded the MediaPipeFaceLink map, then I played it in-Editor

The NvAR option now works on my machine. Thank you again for your help, and thank you for writing the MediaPipe4U plugin.

Jen 😀

**Comment by endink** [comment]
Glad to hear you solved it😄
